specs/fault-proof/stage-one/anchor-state-registry.md

```diff
@@ -0,0 +1,462 @@
+- Integration testing.
+
+## System Invariants
+
+### iASR-001: Games that make correct claims about L2 state can be distinguished from games that do not.
+
+System components that depend on accurate L2 state must be able to trust the ASR to validate games correctly.
+
+#### Impact
+
+**Severity: Critical**
+
+If this invariant is broken, the system will not be able to distinguish between games that make correct claims about L2
+state and games that do not. This could lead to incorrect withdrawals and other problems with L2-to-L1 dependents,
+implying lost funds and lost confidence.
+
+#### Dependencies
+
+- [aFDG-001](#afdg-001-fault-dispute-games-correctly-report-certain-properties)
+- [aDGF-001](#adgf-001-dispute-game-factory-correctly-identifies-the-games-it-created)
+- [aDGF-002](#adgf-002-games-created-by-the-disputegamefactory-will-be-monitored)
+- [aASR-001](#aasr-001-incorrectly-resolving-games-will-be-blacklisted-within-the-dispute-game-finality-delay-period)
+- [aASR-002](#aasr-002-if-a-larger-dispute-game-bug-is-found-all-games-will-be-retired-before-the-first-incorrect-games-dispute-game-finality-delay-period-has-passed)
+- [aASR-003](#aasr-003-the-anchorstateregistry-will-be-correctly-initialized-at-deployment)
+- [aSC-001](#asc-001-superchainconfig-correctly-reports-its-guardian-address)
+
+### iASR-002: Valid withdrawals can be finalized within some bounded amount of time
+
+#### Impact
+
+**Severity: Critical**
+
+If this invariant is broken, withdrawals can be frozen for a long period of time, causing a critical liveness failure.
+
+#### Dependencies
+
+- [aFDG-001](#afdg-001-fault-dispute-games-correctly-report-certain-properties)
+- [aDGF-001](#adgf-001-dispute-game-factory-correctly-identifies-the-games-it-created)
+- [aDGF-002](#adgf-002-games-created-by-the-disputegamefactory-will-be-monitored)
+- [aASR-001](#aasr-001-incorrectly-resolving-games-will-be-blacklisted-within-the-dispute-game-finality-delay-period)
+- [aASR-002](#aasr-002-if-a-larger-dispute-game-bug-is-found-all-games-will-be-retired-before-the-first-incorrect-games-dispute-game-finality-delay-period-has-passed)
+- [aASR-003](#aasr-003-the-anchorstateregistry-will-be-correctly-initialized-at-deployment)
+- [aSC-001](#asc-001-superchainconfig-correctly-reports-its-guardian-address)
+- [iASR-001](#iasr-001-games-that-make-correct-claims-about-l2-state-can-be-distinguished-from-games-that-do-not)
+
+## Component Invariants
+
+### iASR-003: Only "truly" **valid games** will be represented as **valid games**.
+
+When asked for a **valid game** by its dependents, the AnchorStateRegistry will only serve **valid games** representing
+correct L2 state claims.
+
+#### Impact
+
+**Severity: High**
+
+If this invariant is broken, the L1 will have an inaccurate view of L2 state. The OptimismPortal can be tricked into
+finalizing withdrawals based on incorrect state roots, causing loss of funds. Other dependents would also be affected.
+
+#### Dependencies
```

Comment:
Also:
- FDG reports stuff correctly
- People are actually watching
- People will do something if the game resolves incorrectly

Reply:
done in 11b3b54